src/middleware/error-handler.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @typedef {object} ErrorHandlerError
+ * @property {string} err.name
+ * @property {object} [err.json]
+ */
+
+/**
+ * Check if an error is an HttpError
+ * @param {ErrorHandlerError} err
+ * @returns {boolean}
+ */
+function _isHttpError(err) {
+  switch (err.name) {
+    case 'HttpError':
+    case 'BadRequestError':
+    case 'ConflictError':
+    case 'EnhanceYourCalmError':
+    case 'ForbiddenError':
+    case 'IAmATeapotError':
+    case 'InternalServerError':
+    case 'NotFoundError':
+    case 'NotAllowedError':
+    case 'UnauthorizedError':
+    case 'UnprocessableEntityError':
+      return true
+    default:
+      return false
```

Comment:
I didn't think this through when suggesting we remove `instanceof`. Looking at this, I'm thinking we should go back to `instanceof` and make sure any breaking changes are properly versioned as a major bump. `instanceof` will return true for instances of the class and it's parents, giving us more flexibility without having to define all of these manually.